MM2RandoLib/Randomizers/RText.cs

```diff
@@ -347,30 +359,33 @@ public static void PatchCredits(Patch in_Patch, CompanyName in_CompanyName, Rand
                     }
                 }
 
-                for (Int32 j = 0; j < 9; j++)
+                foreach (EWeaponIndex j in EWeaponIndex.All)
                 {
-                    Int32 dmg = RWeaknesses.BotWeaknesses[newIndex, j];
+                    Int32 dmg = RWeaknesses.BotWeaknesses[txtRobos.ElementAt(newIndex).Key][j];
                     sb.Append($"{RText.GetBossWeaknessDamageChar(dmg)} ");
                 }
 
                 String rowString = sb.ToString().Trim();
 
                 for (Int32 j = 0; j < rowString.Length; j++)
                 {
-                    in_Patch.Add(txtRobos[i] + j,
+                    in_Patch.Add(txtRobos.ElementAt(i).Value + j,
```

Comment:
Do you need to use ElementAt or can you use the index operator?

Reply:
This code probably needs to be rewritten. When I was testing last night I discovered that `ElementAt()` is not remotely related to the index operator when using a Dictionary. I was very surprised/horrified to find out that `ElementAt` returns the elements in insertion order. I don't think I caught all the places where I used it.

To answer your question, no because `txtRobots` is type `Dictionary<EBossIndex, Int32>`, but the loop here is `for (Int32 i = 0; i < txtRobos.Count; i++)`

I kind of ran out of steam rewriting everything by the time I got to this part of `RText.cs`.

I can change the outer loop to use the keys of `txtRobots`, but then I have to compromise on the inner loop, like this

```csharp
                for (Int32 j = 0; j < rowString.Length; j++)
                {
                    in_Patch.Add(txtRobos[i] + j,
                        rowString[j].AsCreditsCharacter(),
                        $"Credits robo weakness table Char #{j + i.Offset * rowString.Length}");
                }
```

In particular, I now have to use `i.Offset` in the last line there. Which is probably okay for now. However, I also have to rewrite the loop above it that does the boss component stuff. But I guess it needs more testing anyway.